samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -353,40 +362,54 @@ function Publish-Labs {
                 Import-Module $modulePath
 
                 $labAccount = Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName
-
+    
                 if ($labAccount -eq $null ){
-                    $labAccount = New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                    $labAccount = New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName
                     Write-Host "$LabAccountName lab account created."
                 }
                 else {
                     Write-Host "$LabAccountName lab account found - skipping create."
                 }
 
-                if ($SharedGalleryResourceGroupName -ne $null && $SharedGalleryName -ne $null){
+                if ($SharedGalleryResourceId -ne $null){
 
                     $gallery = $labAccount | Get-AzLabAccountSharedGallery
                     if ($gallery -ne $null) {
                         Write-Host "$LabAccountName lab account already has attached gallery $gallery."
                     }
                     else {
-                        $gallery = Get-AzGallery -ResourceGroupName $SharedGalleryResourceGroupName -Name $SharedGalleryName
+                        $gallery = Get-AzGallery -ResourceId $SharedGalleryResourceId
 
                         if ($gallery -ne $null) {
-                         Write-Host "$SharedGalleryName shared gallery found."
+                         Write-Host "$SharedGalleryResourceId shared gallery found."
                             New-AzLabAccountSharedGallery -LabAccount $labAccount -SharedGallery $gallery
-                            Write-Host "$SharedGalleryName shared gallery attached."
+                            Write-Host "$SharedGalleryResourceId shared gallery attached."
                         }
                         else {
-                            Write-Host "$SharedGalleryName shared gallery not found - skipping attach."
+                            Write-Host "$SharedGalleryResourceId shared gallery not found - skipping attach."
                         }
                     }
+
+                    Write-Host "Enabling images for lab account: $labAccount"
+                    if ($EnableSharedGalleryImages -ne $null)
+                    {
+                        $imageNames = $EnableSharedGalleryImages.Split(',')
+                    }
+
+                    Write-Host "Images to enable: $imageNames"
+                    $images = $labAccount | Get-AzLabAccountSharedImage -EnableState All
+                    foreach ($imageName in $imageNames)
+                    {
+                        $image = $images | Where-Object { $_.name -like $imageName } | Set-AzLabAccountSharedImage -EnableState Enabled
```

Comment:
Should we explicitly set the ones not in the CSV file as disabled, if the field is included?  I'm not sure but that might be the default anyway right after attaching the SIG...

Reply:
Yes I can do that...i think that makes sense to do.